docs/ref/csp.txt

```diff
@@ -0,0 +1,260 @@
+loaded. It plays an important role in a comprehensive :ref:`security strategy
+<security-csp>`.
+
+For configuration instructions in a Django project, see the :ref:`Using CSP
+<using-csp>` documentation. For an HTTP guide about CSP, see the `MDN Guide on
+CSP <https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/CSP>`_.
+
+Overview
+========
+
+The `Content-Security-Policy specification <https://www.w3.org/TR/CSP3/>`_
+defines two complementary headers:
+
+* ``Content-Security-Policy``: Enforces the CSP policy, blocking content that
+  violates the defined directives.
+* ``Content-Security-Policy-Report-Only``: Reports CSP violations without
+  blocking content, allowing for non-intrusive testing.
+
+Each policy is composed of one or more directives and their values, which
+together instruct the browser on how to handle specific types of content.
+
+When the :class:`~django.middleware.csp.ContentSecurityPolicyMiddleware` is
+enabled, Django automatically builds and attaches the appropriate headers to
+each response based on the configured :ref:`settings <csp-settings>`, unless
+they have already been set by another layer.
+
+.. _csp-settings:
+
+Settings
+========
+
+The :class:`~django.middleware.csp.ContentSecurityPolicyMiddleware` is
+configured using the ``SECURE_CSP`` and ``SECURE_CSP_REPORT_ONLY`` settings.
+
+:setting:`SECURE_CSP`
+
+Defines the **enforced Content Security Policy**.
+
+When set, the middleware adds a ``Content-Security-Policy`` HTTP header to
+responses. This instructs the browser to actively block any content that
+violates the configured directives.
+
+.. _content-security-policy-report-only:
+
+:setting:`SECURE_CSP_REPORT_ONLY`
+
+Defines a **report-only Content Security Policy**.
+
+When set, the middleware adds a ``Content-Security-Policy-Report-Only`` HTTP
+header to responses. This allows the browser to monitor and report policy
+violations without blocking content, making it useful for refining a policy.
+
+Most browsers log CSP violations to the developer console and can optionally
+send them to a reporting endpoint. To collect these reports, the ``report-uri``
+directive must be defined (see :ref:`csp-reports` for more details).
+
+.. admonition:: These settings can be used independently or together
+
+   * Use :setting:`SECURE_CSP` alone to enforce a policy that has already been
+     tested and verified.
+   * Use :setting:`SECURE_CSP_REPORT_ONLY` on its own to evaluate a new policy
+     without disrupting site behavior. This mode does not block violations, it
+     only logs them. It's useful for testing and monitoring, but provides no
+     protection against active threats.
+   * Use *both* to maintain an enforced baseline while experimenting with
+     changes. Even for well-established policies, continuing to collect reports
+     reports can help detect regressions, unexpected changes in behavior, or
+     potential tampering in production environments.
+
+Settings structure
```

Comment:
Similar here, I'm not sure if this information is covered (or should be covered) by the settings docs

Reply:
I usually struggle with the difficult balance between having enough details in the settings page but not "too much". Given your commentary feels like this could be moved to the settings page.